src/hw-2/routes.ts

```diff
@@ -0,0 +1,54 @@
+import express from 'express';
+import * as core from 'express-serve-static-core';
+import { validation } from './validation';
+
+import { User } from './types';
+import { UserService } from './user.service';
+
+const router: core.Router = express.Router();
+const userService = new UserService();
+
+router.route('/user/:id')
+    .get((req: express.Request, res: express.Response) => {
+       const user: User | undefined = userService.getUser(req.params.id);
+
+       user
+        ? res.json(user)
+        : res.status(404).json({message: `User with id ${req.params.id} is not found`});
+    })
+    .delete((req: express.Request, res: express.Response) => {
+        if (userService.isUserExist(req.params.id)) {
+            userService.deleteUser(req.params.id);
+
+            res.json({message: `User with id ${req.params.id} is marked as deleted`})
+        } else {
+            res.status(404).json({message: `User with id ${req.params.id} is not found`});
+        }
+    })
+    .patch((req: express.Request, res: express.Response) => {
+        const user: User | undefined = userService.getUser(req.params.id);
+
+        if (user) {
+            userService.updateUser(req.body, req.params.id);
+            
+            res.json({message: `User id ${req.params.id} is updated`});
+        } else {
+            res.status(404).json({message: `User with id ${req.params.id} is not found`});
+        }
+    });
+
+router.route('/user')
+    .post(validation(), (req: express.Request, res: express.Response) => {
+        const user: User = userService.createUser(req.body);
```

Comment:
not sure you need to separate just `createUser` on `create` and `add`. In real world you would have just `createUser` which would do some some in-memory work and then would put this to db, no sure you'll need separated `add` to write that user to db.

Reply:
Fixed.